subprojects/core/src/main/java/org/gradle/internal/progress/TaskOutcomeStatisticsFormatter.java

```diff
@@ -19,23 +19,38 @@
 import org.gradle.api.internal.tasks.TaskStateInternal;
 import org.gradle.api.tasks.TaskState;
 
+import java.util.concurrent.atomic.AtomicInteger;
+
 public class TaskOutcomeStatisticsFormatter {
-    private int executedTasksCount;
-    private int allTasksCount;
+    private AtomicInteger avoidedTasksCount = new AtomicInteger(0);
```

Comment:
I don't think we need thread safety here: https://github.com/gradle/gradle/blob/0c4a1ab113654969fd2679e9e201b4ef75daad71/subprojects/messaging/src/main/java/org/gradle/internal/event/ListenerManager.java#L34-L34